pallets/identity/src/keys.rs

```diff
@@ -244,6 +261,75 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Accepts a primary key rotation.
+    /// Differs from accept_primary_key_rotation in that it will leave the old primary key as a
+    /// secondary key with the permissions specified in the corresponding DirectRotatePrimaryKey authorization
+    /// instead of unlinking the primary key.
+    crate fn base_rotate_primary_key_to_secondary(
```

Comment:
Sorry for spamming comments but thinking about this a bit more `base_rotate_primary_key_to_secondary` should include all steps required in `unsafe_primary_key_rotation` so it can be called instead of duplicating code in this function.

The only differences between `base_rotate_primary_key_to_secondary` and `accept_primary_key_rotation` should be that the old primary key gets added back as a secondary for `base_rotate_primary_key_to_secondary` with appropriate permissions.

Both functions should unlink old primary, unlink new primary **if it is a secondary** and link new primary which are covered by `unsafe_primary_key_rotation` (once it's updated to remove secondary keys)

Reply:
@F-OBrien Why unlink just to re-link to the same identity?  The unlink should only happen if the key is going to be left unlinked when the call finishes.

Reply:
> 

Sorry my terminology probably wasn't correct. I meant remove primary key or remove secondary key when I said unlink. For `base_rotate_primary_key_to_secondary` the old primary key should remain linked to the DID and in both cases if the new primary is an old secondary it can stay linked to the DID. Only in the case of `accept_primary_key_rotation` should the old primary be removed from the DID. I missed that that was included in `unsafe_primary_key_rotation` but looks like @raycar5 already caught it in `rotate_primary_key_and_remove` which uses `common_rotate_primary_key` to reduce the code duplication instead of straight using `unsafe_primary_key_rotation`